pkg/utils/gardener/project.go

```diff
@@ -34,6 +36,24 @@ func ProjectForNamespaceFromReader(ctx context.Context, reader client.Reader, na
 	return &projectList.Items[0], nil
 }
 
+// ProjectForNamespaceFromLister returns the Project responsible for a given <namespace>. It lists all Projects
+// via the given lister, iterates over them and tries to identify the Project by looking for the namespace name
+// in the project spec.
+func ProjectForNamespaceFromLister(projectLister gardencorev1beta1listers.ProjectLister, namespaceName string) (*gardencorev1beta1.Project, error) {
```

Comment:
With this, we have some weird import dependencies and lifted restrictions. Maybe you can move this function to `pkg/apiserver` to restrict this to `gardener-apiserver`? I think all other components use `client.Client` and no listers?

Reply:
I can move it back to `plugin/pkg/utils/`, but I was thinking as `ProjectForNamespaceFromReader` and `ProjectForNamespaceFromLister` are doing pretty much the same with the only difference being the client, they should be implemented in the same package. 

Btw, I firstly found `ProjectForNamespaceFromReader` and worked for some time to introduce `client.Reader` into the apiserver, but then I thought it might not be good idea to introduce controller-runtime. After that I somehow discovered  `ProjectForNamespaceFromLister`. 

So, instead of moving it away from `plugin/pkg/utils`, should I just make an alias to it from `pkg/utis/gardener` so that it is visible next to `ProjectForNamespaceFromReader`?

Reply:
Wouldn't an alias lead to the same import dependency problems? This is my only concern. I agree that it would be nice to have both functions side-by-side (I think this was even the case some time back), but at some point we decided to separate this to get of some "weird" dependencies.

Reply:
The function is moved back at `plugin/pkg/utils`.